diff --git a/node_modules/@abandonware/noble/lib/hci-socket/gatt.js b/node_modules/@abandonware/noble/lib/hci-socket/gatt.js
index d04e53a..f408f72 100644
--- a/node_modules/@abandonware/noble/lib/hci-socket/gatt.js
+++ b/node_modules/@abandonware/noble/lib/hci-socket/gatt.js
@@ -644,13 +644,20 @@ Gatt.prototype.notify = function(serviceUuid, characteristicUuid, notify) {
       var valueBuffer = new Buffer(2);
       valueBuffer.writeUInt16LE(value, 0);
 
-      this._queueCommand(this.writeRequest(handle, valueBuffer, false), function(data) {
+      var writeCb = function(data) {
         var opcode = data[0];
+        console.log('WCOMM', data, serviceUuid, characteristicUuid);
 
         if (opcode === ATT_OP_WRITE_RESP) {
+          console.log('NOTIF', serviceUuid, characteristicUuid);
           this.emit('notify', this._address, serviceUuid, characteristicUuid, notify);
         }
-      }.bind(this));
+        else if (opcode === ATT_OP_ERROR) {
+          console.error("ATT_OP_ERROR, retrying...");
+          this._queueCommand(this.writeRequest(handle, valueBuffer, false), writeCb);
+        }
+      }.bind(this);
+      this._queueCommand(this.writeRequest(handle, valueBuffer, false), writeCb);
     }
   }.bind(this));
 };
